Tue Sep 16 07:16:53 BST 2008  Steve Cotton <steve@s.cotton.clara.co.uk>
  * 400768: Use ints instead of shorts for ID fields (breaks ABI)
  Increases limits on number of packages, versions and descriptions
  from 2**16-1 to 2**32-1.
diff -rN -u old-apt-0.7.14.steve.1/apt-pkg/pkgcache.h new-apt-0.7.14.steve.1/apt-pkg/pkgcache.h
--- old-apt-0.7.14.steve.1/apt-pkg/pkgcache.h	2008-09-16 07:26:58.000000000 +0100
+++ new-apt-0.7.14.steve.1/apt-pkg/pkgcache.h	2008-09-16 07:26:58.000000000 +0100
@@ -214,7 +214,7 @@
    unsigned char InstState;         // Flags
    unsigned char CurrentState;      // State
    
-   unsigned short ID;
+   unsigned int ID;
    unsigned long Flags;
 };
 
@@ -235,7 +235,7 @@
    
    // Linked list
    map_ptrloc NextFile;        // PackageFile
-   unsigned short ID;
+   unsigned int ID;
    time_t mtime;                  // Modification time for the file
 };
 
@@ -272,7 +272,7 @@
    map_ptrloc Size;              // These are the .deb size
    map_ptrloc InstalledSize;
    unsigned short Hash;
-   unsigned short ID;
+   unsigned int ID;
    unsigned char Priority;
 };
 
@@ -289,7 +289,7 @@
    map_ptrloc NextDesc;          // Description
    map_ptrloc ParentPkg;         // Package
 
-   unsigned short ID;
+   unsigned int ID;
 };
 
 struct pkgCache::Dependency

